tests/unit/test_github_client.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Tests for the github_client function."""
+from unittest.mock import MagicMock
+
+import pytest
+from github import GithubException, RateLimitExceededException
+
+from repo_policy_compliance import target_branch_protection
+from repo_policy_compliance.exceptions import GithubClientError
+
+GITHUB_REPOSITORY_NAME = "test/repository"
+GITHUB_BRANCH_NAME = "arbitrary"
+
+
+@pytest.mark.parametrize(
+    "raised_exception, expected_message",
+    [
+        pytest.param(
+            RateLimitExceededException("", "", {}),
+            "Rate Limit Exceeded error",
+            id="github_client rate limit error",
+        ),
+        pytest.param(
+            GithubException("", "", {}), "encountered an error", id="git_client other error"
+        ),
+    ],
+)
+def test_github_error(
+    raised_exception: GithubException,
+    expected_message: str,
+):
+    """
+    arrange: A github repository name and a github_client method that raises exception.
+    act: when target_branch_protection method is called.
+    assert: An expected error is raised with specific error message.
+    """
+    github_client = MagicMock()
+    github_client.get_repo.side_effect = raised_exception
+
+    with pytest.raises(GithubClientError) as error:
+        # target_branch_protection is behind a decorator that injects github_client if not provided
+        target_branch_protection(
```

Comment:
It is actually possible to test this without using `target_branch_protection`, e.g., just pass another function to the decorator like `inject(<some function e.g. a lambda>)()`